hathorlib/base_transaction.py

```diff
@@ -397,14 +397,34 @@ def is_nft_creation_standard(self) -> bool:
         return False
 
     def is_standard(self, std_max_output_script_size: int = settings.PUSHTX_MAX_OUTPUT_SCRIPT_SIZE,
-                    only_standard_script_type: bool = True) -> bool:
+                    only_standard_script_type: bool = True,
+                    max_number_of_data_script_outputs: int = settings.MAX_DATA_SCRIPT_OUTPUTS) -> bool:
         """ Return True is the transaction is standard
         """
+        # First we check if t's an NFT standard
+        # We could remove this because now that we are adding support
+        # for some data script outputs in a transaction, this would
+        # also be considered a standard but if we change our minds
+        # about the data scripts in the future we would need to remember
+        # to add NFT support back, so I'm just keeping this here
         if self.is_nft_creation_standard():
             return True
 
+        # We've discussed to allow any number of Data Script outputs but we decided to
+        # add some restrictions first. Because of that we are not making Data Script a
+        # standard script and we are handling it manually
+        number_of_data_script_outputs = 0
         for output in self.outputs:
             if not output.is_standard_script(std_max_output_script_size, only_standard_script_type):
+                # If not standard then we check if it's a data script with valid size
+                if output.is_script_size_valid(std_max_output_script_size) and output.is_data_script():
+                    # Then we check if it already reached the maximum number of data script outputs
+                    if number_of_data_script_outputs == max_number_of_data_script_outputs:
+                        return False
```

Comment:
There are multiple reasons this method might return `True` or `False`. As it's a library, I feel we shouldn't change the method behavior (e.g., raise exceptions) but we can add an optional `error_list: Optional[List[str]]=None` that will append a list of errors. It seems useful for debugging. What do you think?

To keep the code simpler, we can do the following at the beginning of the method:

```python
if error_list is None:
    error_list: List[str] = []
```

Reply:
I agree with you this would be the best but I tried not to change the return type of the method. Do you feel it's ok to change it and add some breaking changes to this lib version?

If we decide to do that, we can even start raising exceptions.